packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/notifications/RealmDictionaryNotificationsTests.kt

```diff
@@ -0,0 +1,390 @@
+            .directory(tmpDir)
+            .build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (this::realm.isInitialized && !realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    override fun initialElement() {
+        val dataSet = NULLABLE_DICTIONARY_OBJECT_VALUES.mapIndexed { i, value ->
+            Pair(DICTIONARY_KEYS_FOR_NULLABLE[i], value)
+        }
+
+        val container = realm.writeBlocking {
+            copyToRealm(RealmDictionaryContainer()).also {
+                it.nullableObjectDictionaryField.putAll(dataSet)
+            }
+        }
+
+        runBlocking {
+            val channel = Channel<DictionaryChange<*>>(capacity = 1)
+            val observer = async {
+                container.nullableObjectDictionaryField
+                    .asFlow()
+                    .collect { flowDictionary ->
+                        channel.send(flowDictionary)
+                    }
+            }
+
+            // Assertion after empty dictionary is emitted
+            channel.receive().let { dictionaryChange ->
+                assertIs<InitialDictionary<*>>(dictionaryChange)
+
+                assertNotNull(dictionaryChange.map)
+                assertEquals(dataSet.size, dictionaryChange.map.size)
+            }
+
+            observer.cancel()
+            channel.close()
+        }
+    }
+
+    @Test
+    @Suppress("LongMethod")
+    override fun asFlow() {
+        val dataSet = NULLABLE_DICTIONARY_OBJECT_VALUES.mapIndexed { i, value ->
+            Pair(keys[i], value)
+        }
+        val container = realm.writeBlocking {
+            // Create an empty container with empty dictionaries
+            copyToRealm(RealmDictionaryContainer())
+        }
+
+        runBlocking {
+            val channel = Channel<DictionaryChange<*>>(capacity = 1)
+            val observer = async {
+                container.nullableObjectDictionaryField
+                    .asFlow()
+                    .collect { flowDictionary ->
+                        if (flowDictionary !is InitialDictionary<*>) {
+                            channel.send(flowDictionary)
+                        }
+                    }
+            }
+
+            // Assert a single insertion is reported
+            realm.writeBlocking {
+                val queriedContainer = findLatest(container)
+                val queriedDictionary = queriedContainer!!.nullableObjectDictionaryField
+                queriedDictionary[dataSet[0].first] = dataSet[0].second
+            }
+
+            channel.receive().let { dictionaryChange ->
+                assertIs<UpdatedDictionary<*>>(dictionaryChange)
+
+                assertNotNull(dictionaryChange.map)
+                assertEquals(1, dictionaryChange.map.size)
+                dictionaryChange.insertions.let { insertions ->
+                    assertEquals(1, insertions.size)
```

Comment:
We should also check that changes and deleted are 0

Reply:
`changes` is missing and will add it, `deletions` are just two lines down.